implementations/rust/ockam/ockam_transport_uds/src/workers/sender.rs

```diff
@@ -0,0 +1,322 @@
+        // https://github.com/build-trust/ockam/blob/5e5a9ddc557daa2e5183d83fb95a821062c2efcf/implementations/rust/ockam/ockam_transport_tcp/src/workers/sender.rs#L134
+        let rx_addr = Address::random_tagged("UdsRecvProcessor");
+        worker.rx_addr = Some(rx_addr.clone());
+
+        // TODO: @ac 0#UdsSendWorker_tx_addr
+        // in:  0#UdsSendWorker_tx_addr_9  <=  [0#UdsRouter_main_addr_0]
+        // out: n/a
+        let tx_mailbox = Mailbox::new(
+            pair.tx_addr(),
+            Arc::new(AllowAll),
+            // Arc::new(ockam_core::AllowSourceAddress(udsrouter_main_addr)),
+            Arc::new(AllowAll),
+            // Arc::new(ockam_core::DenyAll),
+        );
+
+        // @ac 0#UdsSendWorker_int_addr
+        // in:  0#UdsSendWorker_int_addr_10  <=  [0#UdsRecvProcessor_12]
+        // out: n/a
+        let internal_mailbox = Mailbox::new(
+            worker.internal_addr().clone(),
+            Arc::new(AllowAll),
+            // Arc::new(ockam_core::AllowSourceAddress(rx_addr)),
+            Arc::new(AllowAll),
+            // Arc::new(ockam_core::DenyAll),
+        );
+
+        WorkerBuilder::with_mailboxes(Mailboxes::new(tx_mailbox, vec![internal_mailbox]), worker)
+            .start(ctx)
+            .await?;
+
+        Ok(pair)
+    }
+
+    async fn stop_and_unregister(&self, ctx: &Context) -> Result<()> {
+        self.router_handle.unregister(ctx.address()).await?;
+
+        ctx.stop_worker(ctx.address()).await?;
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl Worker for UdsSendWorker {
+    type Context = Context;
+    type Message = Any;
+
+    /// Connect to the UDS socket.
+    ///
+    /// Spawn a UDS Recceiver worker to processes incoming UDS messages
+    async fn initialize(&mut self, ctx: &mut Self::Context) -> Result<()> {
+        ctx.set_cluster(crate::CLUSTER_NAME).await?;
+
+        let path = match self.peer.as_pathname() {
+            Some(p) => p,
+            None => {
+                debug!("Failed to determine peer path.");
+                self.stop_and_unregister(ctx).await?;
+
+                return Err(TransportError::InvalidAddress.into());
+            }
+        };
+
+        if self.tx.is_none() {
+            let path_display = path.display();
+            debug!(addr = %path_display, "Connecting");
+
+            let connection = match UnixStream::connect(path).await {
+                Ok(c) => {
+                    debug!(addr = %path_display, "Connected");
+                    c
+                }
+                Err(e) => {
+                    debug!(addr = %path_display, err = %e, "Failed to connect");
+                    self.stop_and_unregister(ctx).await?;
+
+                    return Err(TransportError::from(e).into());
+                }
+            };
+
+            let sock = SockRef::from(&connection);
+
+            // TODO: @uds enable an automatic keep alive system
```

Comment:
What is the conclusion from this? Should we enable call `set_keepalive`?

Reply:
So my understanding on this is that socket2, only has the setting  to enable the Unix Domain Socket to allow receive keepalive packets.

However, unlike it's TCP counterpart, it does not have an internal system to automatically send those packets and keep it alive.

I wasn't able to find an existing library that enabled this. So we would need to write a small implementation of this if we wanted it.

However, because the sockets are local to a machine, and less likely to have any issues (e.g. network inconsistencies) I think this low priority.

Reply:
Longstory short, I don't see a reason why we should disable keepalive packets.

However, there won't be any we're sending through this existing implementation.